--- a/test/runtests	2020-12-16 10:12:27.093075638 -0500
+++ b/test/runtests	2020-12-16 10:15:04.924143621 -0500
@@ -7,9 +7,10 @@
 ##     See COPYRIGHT in top-level directory
 ##
 
-prefix=/usr
-exec_prefix=/usr
-bindir=/usr/lib64/mpich/bin
+prefix=/usr/lib64/mpich
+exec_prefix=/usr/lib64/mpich
+bindir=/usr/lib64/mpich/bin
+testsdir=${0%/*}
 mpirun="${bindir}/mpiexec"
 STOPFILE=${MPITEST_STOPTEST:-"$HOME/.stopmpichtests"}
 #
@@ -29,7 +30,7 @@
 	;;
         -margs=*)
 	shift
-	margs=`echo $arg | sed 's/-margs=//'`
+	margs=$(echo $arg | sed 's/-margs=//')
 	args="$args $margs"
 	;;
 	-echo)
@@ -52,7 +53,7 @@
 	subset_only=1
 	;;
 	-fname=*)
-	FILENAME=`echo $arg|sed 's/-*fname=//'`
+	FILENAME=$(echo $arg|sed 's/-*fname=//')
 	;;
 	-atend)
 	shift
@@ -87,7 +88,7 @@
         echo "Found stopfile; exiting"
         exit 0
     fi
-    if [ ! -x $1 ] ; then
+    if [ ! -x $testsdir/$1 ] ; then
 	$MAKE $1
         if [ ! -x $1 ] ; then 
 	    echo "Could not build executable $1; aborting tests"
@@ -105,9 +106,9 @@
 OutTime() {
     if [ $quiet = 0 ] ; then
 	if [ -z "$hostname" ] ; then
-	    hostname=`hostname`
+	    hostname=$(hostname)
 	fi
-	d=`date`
+	d=$(date)
 	echo "$hostname : $d"
     fi
 }
@@ -154,13 +155,13 @@
       MakeExe simple
       rm -rf $FILENAME*
       echo "**** Testing ${1}.c ****"
-      mpirun -np 4 ./${1} -fname $FILENAME "$@" >>${1}.out 2>1
+      mpirun -np 4 $testsdir/${1} -fname $FILENAME "$@" >>${1}.out 2>1
       CheckOutput ${1} 
       CleanExe ${1}
 }
 
 # If the programs are not available, run make.
-if [ ! -x simple -a $makeeach = 0 ] ; then
+if [ ! -x $testsdir/simple -a $makeeach = 0 ] ; then
     $MAKE default
 fi
 #
@@ -175,7 +176,7 @@
 MakeExe simple
 \rm -f $FILENAME*
 echo '**** Testing simple.c ****'
-$mpirun -np 4 ./simple -fname $FILENAME
+$mpirun -np 4 $testsdir/simple -fname $FILENAME
 # CheckOutput simple
 CleanExe simple
 #
@@ -185,7 +186,7 @@
 MakeExe async
 \rm -f $FILENAME*
 echo '**** Testing async.c ****'
-$mpirun -np 4 ./async -fname $FILENAME
+$mpirun -np 4 $testsdir/async -fname $FILENAME
 # CheckOutput async
 CleanExe async
 #
@@ -195,7 +196,7 @@
 MakeExe async-multiple
 \rm -f $FILENAME*
 echo '**** Testing async-multiple.c ****'
-$mpirun -np 4 ./async-multiple -fname $FILENAME
+$mpirun -np 4 $testsdir/async-multiple -fname $FILENAME
 # CheckOutput async-multiple
 CleanExe async-multiple
 #
@@ -208,7 +209,7 @@
 	echo '**** Testing atomicity.c ****'
 	# Atomicity test recommends at least 8 processes (separate processors 
 	# even better)
-	$mpirun -np 4 ./atomicity -fname $FILENAME
+	$mpirun -np 4 $testsdir/atomicity -fname $FILENAME
 	# CheckOutput atomicity
 	CleanExe atomicity
 	#
@@ -219,7 +220,7 @@
 MakeExe coll_test
 \rm -f $FILENAME*
 echo '**** Testing coll_test.c ****'
-$mpirun -np 4 ./coll_test -fname $FILENAME
+$mpirun -np 4 $testsdir/coll_test -fname $FILENAME
 # CheckOutput coll_test
 CleanExe coll_test
 #
@@ -229,7 +230,7 @@
 MakeExe excl
 \rm -f $FILENAME*
 echo '**** Testing excl.c ****'
-$mpirun -np 4 ./excl -fname $FILENAME
+$mpirun -np 4 $testsdir/excl -fname $FILENAME
 # CheckOutput excl
 CleanExe excl
 #
@@ -239,7 +240,7 @@
 MakeExe file_info
 \rm -f $FILENAME*
 echo '**** Testing file_info.c ****'
-$mpirun -np 4 ./file_info -fname $FILENAME
+$mpirun -np 4 $testsdir/file_info -fname $FILENAME
 # CheckOutput file_info
 CleanExe file_info
 #
@@ -249,7 +250,7 @@
 MakeExe i_noncontig
 \rm -f $FILENAME*
 echo '**** Testing i_noncontig.c ****'
-$mpirun -np 2 ./i_noncontig -fname $FILENAME
+$mpirun -np 2 $testsdir/i_noncontig -fname $FILENAME
 # CheckOutput i_noncontig
 CleanExe i_noncontig
 #
@@ -259,7 +260,7 @@
 MakeExe noncontig
 \rm -f $FILENAME*
 echo '**** Testing noncontig.c ****'
-$mpirun -np 2 ./noncontig -fname $FILENAME
+$mpirun -np 2 $testsdir/noncontig -fname $FILENAME
 # CheckOutput noncontig
 CleanExe noncontig
 #
@@ -269,7 +270,7 @@
 MakeExe noncontig_coll
 \rm -f $FILENAME*
 echo '**** Testing noncontig_coll.c ****'
-$mpirun -np 2 ./noncontig_coll -fname $FILENAME
+$mpirun -np 2 $testsdir/noncontig_coll -fname $FILENAME
 # CheckOutput noncontig_coll
 CleanExe noncontig_coll
 #
@@ -279,30 +280,30 @@
 MakeExe noncontig_coll2
 \rm -f $FILENAME*
 echo '**** Testing noncontig_coll2.c ****'
-$mpirun -np 4 ./noncontig_coll2 -fname $FILENAME
+$mpirun -np 4 $testsdir/noncontig_coll2 -fname $FILENAME
 # CheckOutput noncontig_coll2
 CleanExe noncontig_coll2
 MakeExe aggregation1
 echo '**** Testing aggregation1 ****'
-$mpirun -np 4 ./aggregation1 -h -f $FILENAME
+$mpirun -np 4 $testsdir/aggregation1 -h -f $FILENAME
 # CheckOutput aggregation1
 CleanExe aggregation1
 MakeExe aggregation2
 echo '**** Testing aggregation2 ****'
-$mpirun -np 4 ./aggregation2 $FILENAME
+$mpirun -np 4 $testsdir/aggregation2 $FILENAME
 # CheckOutput aggregation2
 CleanExe aggregation2
 MakeExe hindexed
 echo '**** Testing hindexed ****'
-$mpirun -np 4 ./hindexed $FILENAME
+$mpirun -np 4 $testsdir/hindexed $FILENAME
 # CheckOutput hindexed
 CleanExe hindexed
 #
 #echo '**** Testing write_all_test (run 1)****'
-#$mpirun -np 4 ./write_all_test -nzp 2 -zplace 2 -nzw 2 -naw 2 -size 100 \
+#$mpirun -np 4 $testsdir/write_all_test -nzp 2 -zplace 2 -nzw 2 -naw 2 -size 100 \
 #	-offm 8 -hints romio_cb_write enable -fname $FILENAME
 #echo '**** Testing write_all_test (run 2)****'
-#$mpirun -np 4 ./write_all_test -nzp 2 -zplace 2 -nzw 2 -naw 2 -size 100 \
+#$mpirun -np 4 $testsdir/write_all_test -nzp 2 -zplace 2 -nzw 2 -naw 2 -size 100 \
 #	-offm 4 -hints romio_cb_write enable -fname $FILENAME
 #CleanExe write_all_test
 OutTime
@@ -312,7 +313,7 @@
 	MakeExe misc
 	\rm -f $FILENAME*
 	echo '**** Testing misc.c ****'
-	$mpirun -np 4 ./misc -fname $FILENAME
+	$mpirun -np 4 $testsdir/misc -fname $FILENAME
 	# CheckOutput misc
 	CleanExe misc
 	#
@@ -324,7 +325,7 @@
 	MakeExe shared_fp
 	\rm -f $FILENAME*
 	echo '**** Testing shared_fp.c ****'
-	$mpirun -np 4 ./shared_fp -fname $FILENAME
+	$mpirun -np 4 $testsdir/shared_fp -fname $FILENAME
 	# CheckOutput shared_fp
 	CleanExe shared_fp
 	#
@@ -334,7 +335,7 @@
 	MakeExe ordered_fp
 	\rm -f $FILENAME*
 	echo '**** Testing ordered_fp.c ****'
-	$mpirun -np 4 ./ordered_fp -fname $FILENAME
+	$mpirun -np 4 $testsdir/ordered_fp -fname $FILENAME
 	CleanExe ordered_fp
 	#
 	OutTime
@@ -344,7 +345,7 @@
 MakeExe split_coll
 \rm -f $FILENAME*
 echo '**** Testing split_coll.c ****'
-$mpirun -np 4 ./split_coll -fname $FILENAME
+$mpirun -np 4 $testsdir/split_coll -fname $FILENAME
 # CheckOutput split_coll
 CleanExe split_coll
 #
@@ -354,7 +355,7 @@
 MakeExe psimple
 \rm -f $FILENAME*
 echo '**** Testing psimple.c ****'
-$mpirun -np 4 ./psimple -fname $FILENAME
+$mpirun -np 4 $testsdir/psimple -fname $FILENAME
 # CheckOutput psimple
 CleanExe psimple
 #
@@ -364,7 +365,7 @@
 MakeExe error
 \rm -f $FILENAME*
 echo '**** Testing error.c ****'
-$mpirun -np 1 ./error -fname $FILENAME
+$mpirun -np 1 $testsdir/error -fname $FILENAME
 # CheckOutput error
 CleanExe error
 #
@@ -375,7 +376,7 @@
 \rm -f $FILENAME*
 echo '**** Testing status.c ****'
 # Some systems have a status program.
-$mpirun -np 1 ./status -fname $FILENAME
+$mpirun -np 1 $testsdir/status -fname $FILENAME
 # CheckOutput status
 CleanExe status
 testfiles="$testfiles types_with_zeros.out"
@@ -383,18 +384,18 @@
 MakeExe types_with_zeros
 \rm -f $FILENAME*
 echo '**** Testing types_with_zeros ****'
-$mpirun -np 2 ./types_with_zeros $FILENAME
+$mpirun -np 2 $testsdir/types_with_zeros $FILENAME
 CleanExe types_with_zeros
 testfiles="$testfiles darray_read.out"
 \rm -f darray_read.out
 MakeExe darray_read
 \rm -f $FILENAME*
 echo '**** Testing darray_read ****'
-$mpirun -np 4 ./darray_read $FILENAME
+$mpirun -np 4 $testsdir/darray_read $FILENAME
 CleanExe darray_read
 MakeExe syshints
 echo '**** Testing syshints ****'
-$mpirun -np 1 ./syshints $srcdir/test_hintfile
+$mpirun -np 1 $testsdir/syshints $testsdir/test_hintfile
 CleanExe syshints
 
 #
@@ -408,7 +409,7 @@
 	    MakeExe fmisc
 	    \rm -f $FILENAME*
 	    echo '**** Testing fmisc.f ****'
-	    $mpirun -np 4 ./fmisc -fname $FILENAME
+	    $mpirun -np 4 $testsdir/fmisc -fname $FILENAME
 	    # CheckOutput fmisc
 	    CleanExe fmisc
 	    #
@@ -419,7 +420,7 @@
     MakeExe fcoll_test
     \rm -f $FILENAME*
     echo '**** Testing fcoll_test.f ****'
-    $mpirun -np 4 ./fcoll_test -fname $FILENAME
+    $mpirun -np 4 $testsdir/fcoll_test -fname $FILENAME
     # CheckOutput fcoll_test
     CleanExe fcoll_test
 #
@@ -429,7 +430,7 @@
     MakeExe pfcoll_test
     \rm -f $FILENAME*
     echo '**** Testing pfcoll_test.f ****'
-    $mpirun -np 4 ./pfcoll_test -fname $FILENAME
+    $mpirun -np 4 $testsdir/pfcoll_test -fname $FILENAME
     # CheckOutput pfcoll_test
     CleanExe pfcoll_test
 #
@@ -438,7 +439,7 @@
 #
 else
     # Just run checks
-    testfiles=`echo *.out`
+    testfiles=$(echo *.out)
     if test "$testfiles" = "*.out" ; then
 	echo "No output files remain from previous test!"
 	exit 1
@@ -450,14 +451,14 @@
 # rm -f iotests.diff
 # nodiff=1
 # for file in $testfiles ; do
-#     stdfile="${srcdir}/std/`basename $file .out`.std"
+#     stdfile="${srcdir}/std/$(basename $file .out).std"
 #     # if basename is sendrecv or isndrcv, then we may want to test 
 #     # with .std2 as well.  We should really separate out the long double
 #     # tests ...
 #     if [ -s $stdfile ] ; then
 # 	stdfile2=${stdfile}2
 #         # Handle Fortran systems that generate stop statements
-# 	bfile="`basename $file .out`.tout"
+# 	bfile="$(basename $file .out).tout"
 #         rm -f ${bfile}
 # 	grep -v 'FORTRAN STOP' ${file} > ${bfile}
 #         if diff -b $bfile $stdfile > /dev/null ; then
@@ -467,18 +468,18 @@
 # 	    if diff -b $bfile $stdfile2 > /dev/null ; then
 # 	        true
 #             else
-# 	        echo "Differences in `basename $file .out`" >> iotests.diff
+# 	        echo "Differences in $(basename $file .out)" >> iotests.diff
 # 	        diff -b $bfile $stdfile >> iotests.diff
 # 	        nodiff=0
 # 	    fi
 #         else
-# 	    echo "Differences in `basename $file .out`" >> iotests.diff
+# 	    echo "Differences in $(basename $file .out)" >> iotests.diff
 # 	    diff -b $bfile $stdfile >> iotests.diff
 # 	    nodiff=0
 # 	fi
 #         rm -f ${bfile}
 #     else
-#         echo "Can not find file $stdfile to compare against for test `basename $file .out`"
+#         echo "Can not find file $stdfile to compare against for test $(basename $file .out)"
 # 	nodiff=0
 #     fi
 # done
